### DISCUSSION TITLE
Best Practices in matching MacOS/OS X keys with Linux keys

### DISCUSSION THREAD

**Discussion by jasoncorso** [pqrs-org/Karabiner-Elements#4058] [General]
I switch back and forth between MacOS and Linux (Debian) all the time.  I am increasingly frustrated with different keystrokes for common things.  The differing semantic of Control and Command keys is a particular nuisance.  So, I'm in search of best practices for using Karabiner to bring these into alignment.  Thoughts.  (FWIW, On Linux, I use Debian/i3.). Thank you in advance.

**Comment by peterjc** [comment]
You can swap control and command on macOS (without Karabiner Elements), or could probably do the same on Linux. That might help.

I currently mostly work from macOS but often in the terminal, and often ssh’d into a Linux server. Some terminals let you swap control and command with the terminal. The idea being in macOS you use command+… with the prime positioned modifier key on an Apple keyboard - but have that same key behave like Ctrl within the terminal (eg emacs shortcuts).

I think could use KE to swap the modifiers depending on the currently active application (eg your terminal or anything else)? See https://karabiner-elements.pqrs.org/docs/json/complex-modifications-manipulator-definition/conditions/frontmost-application/
